common/src/chain/config.rs

```diff
@@ -199,4 +200,18 @@ mod tests {
         assert_ne!(config.magic_bytes(), mainnet.magic_bytes(),);
         assert_ne!(config.version(), mainnet.version(),);
     }
+
+    #[test]
+    fn chain_type_names() {
+        use strum::VariantNames;
+
+        assert_eq!(&ChainType::Mainnet.to_string(), "mainnet");
+        assert_eq!(&ChainType::Testnet.to_string(), "testnet");
+
+        assert_eq!(ChainType::VARIANTS.len(), 4, "Unexpected number of chain types");
```

Comment:
Is there a reason this is necessary?

Reply:
Just to check the external crate generates the variants, probably too paranoid on my part